lib/solReleases.js

```diff
@@ -0,0 +1,40 @@
+const map = require('../lib/mapping.json');
+const rp = require('request-promise');
+
+function getCompilerVersion(parsedData) {
+    return new Promise(async function(resolve, reject){
+        if(parsedData[0].hasOwnProperty('start_version') == false) {
+            reject(new Error('No Pragma specified !!!'))
+        } else {
+            var pragmaVersion = parsedData[0].start_version.version
+            if(parsedData[0].type == 'PragmaStatement' && Object.keys(map.compilers).indexOf(parsedData[0].start_version.version) > -1){
+                compiler = map.compilers[parsedData[0].start_version.version];
+                resolve(compiler);
+            } else {
+                var options = {
+                    method: 'GET',
+                    uri: 'https://api.github.com/repos/ethereum/solidity/releases',
+                    headers: {'user-agent': 'node.js'}
+                };
+            
+                rp(options)
+                .then(function (result) {
+                    let data = JSON.parse(result)
+                        let index = data.findIndex(obj => obj.tag_name == 'v'+pragmaVersion)
+                        if(index < 0) {
+                            reject(new Error('Unsupported Compiler Version/No Pragma'));
```

Comment:
`No Pragma` check has been implemented above already. We can remove this text from error message.